rust/datafusion/src/physical_plan/functions.rs

```diff
@@ -447,6 +446,26 @@ pub fn return_type(
     }
 }
 
+#[cfg(feature = "crypto_expressions")]
+macro_rules! invoke_if_crypto_expressions_feature_flag {
+    ($FUNC:ident, $NAME:expr) => {{
+        use crate::physical_plan::crypto_expressions;
+        crypto_expressions::$FUNC
+    }};
+}
+
+#[cfg(not(feature = "crypto_expressions"))]
```

Comment:
I think you can put the `#[cfg(feature)]` inside the macro so you dont have to maintain two funtions
What do you think?

```
macro_rules! invoke_if_crypto_expressions_feature_flag {
    ($FUNC:ident, $NAME:expr) => {
        #[cfg(not(feature = "crypto_expressions"))]
        {
            |_: &[ColumnarValue]| -> Result<ColumnarValue> {
                Err(DataFusionError::Internal(format!(
                    "function {} requires compilation with feature flag: crypto_expressions.",
                    $NAME
                )))
            }
        }

        #[cfg(feature = "crypto_expressions")]
        {
            use crate::physical_plan::crypto_expressions;
            crypto_expressions::$FUNC
        }
    };
}
```

Reply:
@seddonm1 That's odd, because before posting that I did a small test and it did compile. 

```rust
#[cfg(feature = "rand_calc")]
use playground::myrand;

macro_rules! testing_something {
    ($name:expr) => {
        #[cfg(not(feature = "rand_calc"))]
        {
            println!("No feature: {}", $name);
        }

        #[cfg(feature = "rand_calc")]
        {
            println!("Feature: {}", $name);
        }
    };
}

fn main() {
    let a = 3.0001f64.to_le_bytes();
    println!("{:?}", a);

    #[cfg(feature = "rand_calc")]
    {
        let res = myrand::with_rand::create_rnd();
        println!("{:?}", res);
    }

    testing_something!("Name");
}
```

I should have tested it with your code to see if it would compile as well